qa/tasks/cephadm.py

```diff
@@ -100,6 +100,38 @@ def update_archive_setting(ctx, key, value):
         yaml.safe_dump(info_yaml, info_file, default_flow_style=False)
 
 
+@contextlib.contextmanager
+def nvmeof_context(ctx, config):
+    source_host = config.get('source')
+    target_host = config.get('target')
+    nvmeof_service = config.get('service')
+    assert source_host
+    assert target_host
+    assert nvmeof_service
+    (remote,) = ctx.cluster.only(source_host).remotes.keys()
+    ip_address = remote.ip_address
+    gateway_addr = remote.sh(f"systemctl list-units | grep {nvmeof_service} | awk '{{print $1}}' | xargs echo | cut -d '@' -f2- | sed 's/.\{{8\}}$//'",
```

Comment:
Other functions in this file use `remote.run`. It would be more consistent to use remote.run, no?
This whole pipeline seems unclear to me, so I tried to run it locally and what it returned was certainly not an address. Additionaly, the `xargs echo` bit seems redundant.

Reply:
Instead of `gateway_addr`, maybe `gateway_name` would be a better variable name. 
Also I'll use `remote.run` instead and remove `xargs echo` part which is indeed unnecessary.